### ISSUE TITLE
Options for public use

### ISSUE THREAD

**Issue by Androkai** [LINKIWI/modern-paste#26]
I miss options to limit non-registered users and automatic deletion of old pastes to ensure system stability under heavy usage.

Some points I think about:
- max expiration date for pastes of non-registered users
- automatic deletion of oldest pastes from non-registered users then defined quota is reached
- file-uploads only for registered users

Btw: Very good paste system, clear look and modern design + very stable, good job!

**Comment by nekohayo** [comment]
+1, I want a paste system where I can tell it to not show up publicly in the list of recent pastes (people need the URL to access it) and where it self-destructs after a user-specified amount of time (1h, 1 day, 1 week, 1 month). If those options are not exposed in the paste UI, I'm left guessing that it publicly lists and retains the paste by default, which makes me not use it (unlike the previous fpaste incarnation).

**Comment by LINKIWI** [comment]
> I want a paste system where I can tell it to not show up publicly in the list of recent pastes (people need the URL to access it)

I agree, this is a possible enhancement that can be added to the project.

> where it self-destructs after a user-specified amount of time (1h, 1 day, 1 week, 1 month)

This is currently supported as the "Paste Expiration" option.
